internal/bminventory/inventory_v2_handlers.go

```diff
@@ -575,57 +575,40 @@ func (b *bareMetalInventory) generateImageDownloadURL(ctx context.Context, infra
 	if err != nil {
 		return "", nil, err
 	}
-	return b.signURL(ctx, infraEnvID, urlString, imageTokenKey)
+	urlString, err = b.signURL(ctx, infraEnvID, urlString, imageTokenKey)
+	if err != nil {
+		return "", nil, err
+	}
+	expiresAt, err := common.ParseExpirationFromURL(urlString)
```

Comment:
Maybe this would be better in `gencrypto` as it's doing token manipulation.

Reply:
Moved it there, thank you!